src/main/java/com/epam/bookstoreservice/service/impl/BookstoreServiceImpl.java

```diff
@@ -144,29 +115,61 @@ public List<BookResponseDTO> getBooksByCategoryAndKeyWord(String category, Strin
 
     @Override
     public Integer getNumberOfBooksSoldPerCategoryAndKeyword(String category, String keyword) {
-        List<BookResponseDTO> bookEntityListFindByCategoryAndKeyWord = getBooksByCategoryAndKeyWordUtil(category, keyword);
 
-        return bookEntityListFindByCategoryAndKeyWord
+        return getBooksByCategoryAndKeyWordUtil(category, keyword)
                 .stream().mapToInt(BookResponseDTO::getSold).sum();
     }
 
 
     private List<BookResponseDTO> getBooksByCategoryAndKeyWordUtil(String category, String keyword) {
-        int bookId = -1;
-        try {
-            bookId = Integer.parseInt(keyword);
-        } catch (Exception ex) {
-            logger.info(ex.getMessage());
-        }
-        List<BookEntity> bookEntityListFindByCategoryAndKeyword = bookDao.findByCategoryAndKeyword(category, bookId, keyword);
+
+        List<BookEntity> bookEntityListFindByCategoryAndKeyword = bookDao
+                .findByCategoryAndKeyword(category, getBookIdWhenKeywordIsNumber(keyword), keyword);
 
         if (bookEntityListFindByCategoryAndKeyword.isEmpty()) {
             throw new BookNotFoundException();
         }
 
         return bookEntityListFindByCategoryAndKeyword
                 .stream()
-                .map(bookRequestDtoToBookEntity::entityToResponseDto).collect(Collectors.toList());
+                .map(bookDtoAndBookEntityMapper::entityToResponseDto).collect(Collectors.toList());
+    }
+
+    private void addBookEntityTotalCount(BookRequestDTO bookRequestDTO, BookEntity bookEntityFindById) {
+        Integer totalCount = bookRequestDTO.getTotalCount() + bookEntityFindById.getTotalCount();
+        bookEntityFindById.setTotalCount(totalCount);
+    }
+
+
+    private BookResponseDTO sellBookByIdAndSalesNumber(Integer id, Integer salesNumber) {
+
+        BookEntity bookEntityFindById = getBookEntityFindByIdIfExist(id);
+
+        if (bookEntityFindById.getTotalCount() - salesNumber < 0) {
+            throw new InsufficientInventoryException();
+        }
+
+        bookEntityFindById.setSold(bookEntityFindById.getSold() + salesNumber);
+        bookEntityFindById.setTotalCount(bookEntityFindById.getTotalCount() - salesNumber);
+
+        return bookDtoAndBookEntityMapper.entityToResponseDto(bookDao.save(bookEntityFindById));
+    }
+
+    private BookEntity getBookEntityFindByIdIfExist(Integer id) {
+        Optional<BookEntity> bookFindById = bookDao.findById(id);
+
+        return bookFindById.orElseThrow(BookNotFoundException::new);
+    }
+
+
+    private int getBookIdWhenKeywordIsNumber(String keyword) {
+        int bookId = -1;
```

Comment:
This value is not specific. If the keyword is invalid, you could raise an exception